src/irmin-git/irmin_git.ml

```diff
@@ -154,7 +156,43 @@ module Make_private
   end
   module XContents = struct
     include AO (GitContents)
-    module Val = C
+    module Val = struct
+      include C
+
+      let encode_bin buf off (t:t) =
+        Log.debug (fun l -> l "Content.encode_bin");
+        let blob = G.Value.Blob.of_string (Irmin.Type.encode_bin C.t t) in
+        match Raw.to_raw (G.Value.blob blob) with
+        | Error _ -> assert false
+        | Ok s    ->
+          let len = String.length s in
```

Comment:
@dinosaure here `len` should probably be the read bits instead of the full buffer lenght.